File: problem_3/prime.f

!
! Project Euler Problem 3: Largest Prime Factor
!
! D. C. Groothuizen Dijkema - November, 2019
!

subroutine run
  implicit none
  integer :: itr
  integer(kind=8) :: val
  integer(kind=8),parameter :: n=600851475143

  val=n
  itr=2

  do while(itr*itr<val)
    if (mod(val,itr).eq.0) then
      val=val/itr
    else
      itr=itr+1
    end if
  end do

  if (val>itr) then
    itr=val
  end if

  print *,"largest prime factor of ",n," is:",itr

end subroutine run

program prime
  implicit none
  integer :: beginning,end,rate

  call system_clock(beginning,rate)
  call run
  call system_clock(end)
  print *,"Elapsed time:",real(end-beginning)/real(rate)

end program

File: problem_3/prime.cpp

//
// Project Euler Problem 3: Largest prime factor
//
// D. C. Groothuizen Dijkema - November, 2019
//

#include <chrono>
#include <iostream>

void run(void)
{
  const long long n=600851475143;
  long long val=n;
  int itr=2;

  while(itr*itr<val)
  {
    if (val%itr==0) { val/=itr; }
    else { ++itr; }
  }

  if (val>itr) { itr=val; }

  std::cout << "The largest prime factor of " << n << " is " << itr << std::endl;
}

int main(void)
{
  std::chrono::time_point<std::chrono::steady_clock> start = std::chrono::high_resolution_clock::now();
  run();
  std::chrono::time_point<std::chrono::steady_clock> finish = std::chrono::high_resolution_clock::now();
  std::chrono::duration<double> elapsed=finish-start;
  std::cout << "Elapsed time: " << elapsed.count() << std::endl;

  return 0;
}
